base/jthread.hpp

```diff
@@ -123,6 +124,13 @@ class this_stoppable_thread {
   friend jthread MakeStoppableThread(Function&& f, Args&&... args);
 };
 
+#define RETURN_IF_STOPPED                                                     \
```

Comment:
I am wondering if this should take (optionally) a parameter to let the caller propagate a useful message.

I am also wondering if our macro names should systematically start with `PRINCIPIA_`.